cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -167,6 +168,12 @@ func NewStaticAutoscaler(
 		ignoredTaints[taintKey] = true
 	}
 
+	unwantedTaints := make(taints.TaintKeySet)
```

Comment:
Can we maybe move this + ignored taints above to a helper function? static_autoscaler.go is already huge.

Also, I'm wondering - if we move it to taints package, maybe we could just have a single variable that represents all taint-related config and pass that around rather than adding more parameters all over the place? If we ever expected 3 TaintKeySets that would be a no-brainer to me, with just 2 I'm not sure if it's worth the effort. I don't really have strong feelings either way so take this part just as a suggestion and feel free to ignore it if you disagree.

Reply:
Added TaintConfig to encapsulate both ignore and status taints.